api/base.proto

```diff
@@ -149,3 +150,16 @@ message ConfigSource {
     AggregatedConfigSource ads = 3;
   }
 }
+
+// Configuration for low-level socket in listeners and clusters. If the configuration is empty,
+// a default low-level socket implementation and configuration will be chosen based on the
+// platform and existence of tls_context.
+message LowLevelSocket {
```

Comment:
I made this name up while brainstorming the overall problem - it wasn't necessarily meant to be the final name. This is probably our last chance to change it here and in Envoy. Do we like this name?

Reply:
No strong feelings, but some names to riff on (some are problematic as is) `SocketFamily`, `TransportSocket`, `SocketConfig`, `SocketWrapper`, ...

Reply:
I think I like `TransportSocket` personally over `LowLevelSocket`, but I don't really care one way or the other. I mostly just wanted to say that we don't need to do this name because of what I said in the related PR.

Reply:
`TransportSocket` or `LowLevelSocket` are both fine, IMHO.

Reply:
I prefer TransportSocket, but either is fine

Reply:
OK @lizan you pick between LowLevelSocket or TransportSocket!

Reply:
I'm fine either way, so I'll update this to `TransportSocket` soon.